TASK: Allow new Driver registration by Settings

dfeyer opened the pull request:
This change allow new driver registration. It the foundation to support Vips Imagine driver build by the Rokka team https://packagist.org/packages/rokka/imagine-vips

dfeyer committed:
TASK: Allow new Driver registration by Settings

dfeyer committed:
TASK: PSR2 fixes

dfeyer committed:
TASK: Add StyleCI configuration

kitsunet reviewed (approved):
Looks good, small nitpick woudl be the buildin functions whcih we usually don't prefix with `\`

kdambekalns commented on Classes/AbstractImagineFactory.php:
```diff
@@ -1,4 +1,5 @@
 <?php
+
```
We require `namespace` without leading blank lines.

kdambekalns committed:
TASK: Code style tweak & change exception code

Changes the exception code, since this is effecticely a new exception (same place,
but slightly different cause and certainly different solution.)
